build.gradle.kts

```diff
@@ -111,17 +111,36 @@ fun Project.setupJacoco() {
         classDirectories.setFrom(
             fileTree("$buildDir/intermediates/classes/debug") {
                 exclude(coverageExclusions)
+            } + fileTree("$buildDir/intermediates/classes/release") {
+                exclude(coverageExclusions)
             } + fileTree("$buildDir/tmp/kotlin-classes/debug") {
                 exclude(coverageExclusions)
+            } + fileTree("$buildDir/tmp/kotlin-classes/release") {
+                exclude(coverageExclusions)
             }
         )
         reports {
             html.required.set(true)
             xml.required.set(true)
         }
-
         sourceDirectories.setFrom(files("$projectDir/src/main"))
-        executionData.setFrom(files("$buildDir/outputs/unit_test_code_coverage/debugUnitTest/testDebugUnitTest.exec"))
+
+        /**
+         * We enable androidTests coverage for debug builds in SpeziBaseConfigConventionPlugin.
+         * There is however an issue when generating jacoco report of unit and android tests
+         * as the classes are already instrumented from the former test type. We follow the solution
+         * from this SO answer: https://stackoverflow.com/a/68481159
+         * We use release execution data for unit test coverage, and debug data for androidTests
+         */
+        executionData.setFrom(
+            files(
+                "$buildDir/outputs/unit_test_code_coverage/releaseUnitTest/releaseDebugUnitTest.exec",
+                "$buildDir/outputs/code_coverage/debugAndroidTest/connected/*.ec"
```

Comment:
The initial issue that you faced, was because the `connected.ec` file, which jacoco uses to create the code coverage reports was not being created. For that it is needed that we explicitly enable `enableAndroidTestCoverage` (added in `SpeziBaseConfigConventionPlugin`). Then I faced the issue in the comment and followed the solution of linked answer.

Reply:
Amazing; thank you @eldcn; this looks great and I looked at the diff to learn what you have changed, very nice! 👍